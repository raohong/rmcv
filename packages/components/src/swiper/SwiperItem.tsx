import React from 'react';
import { animated } from '@react-spring/web';
import classNames from 'classnames';
import type { SwiperItemProps } from './interface';
import { useConfigContext } from '../config-provider';

export const SwiperItemSymbol = Symbol('swiper-item');

const SwiperItem: React.FC<SwiperItemProps> = ({
  children,
  className,
  ...rest
}) => {
  const { getPrefixCls } = useConfigContext();

  return (
    <animated.div
      className={classNames(getPrefixCls('swiper-item'), className)}
      {...rest}
    >
      {children}
    </animated.div>
  );
};

// @ts-ignore
SwiperItem[SwiperItemSymbol] = true;

export default SwiperItem;
