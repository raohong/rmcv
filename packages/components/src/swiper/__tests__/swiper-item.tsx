import { render } from '@testing-library/react';
import SwiperItem from '../SwiperItem';

it('render correctly', () => {
  const tree = render(<SwiperItem />);

  expect(tree.asFragment()).toMatchSnapshot();
});
