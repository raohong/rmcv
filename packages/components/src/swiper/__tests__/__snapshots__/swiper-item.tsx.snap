// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render correctly 1`] = `
<DocumentFragment>
  .emotion-0 {
  box-sizing: border-box;
  width: 100%;
  position: relative;
  overflow: hidden;
}

<div
    class=" emotion-0"
  />
</DocumentFragment>
`;
